src/Common/MemoryTracker.cpp

```diff
@@ -173,29 +173,36 @@ void MemoryTracker::alloc(Int64 size)
     }
 
 #ifdef MEMORY_TRACKER_DEBUG_CHECKS
-    if (unlikely(_memory_tracker_always_throw_logical_error_on_allocation))
+    if (unlikely(_memory_tracker_always_throw_logical_error_on_allocation) && throw_if_memory_exceeded)
```

Comment:
Looks like this should check for `throw_if_memory_exceeded`? since this is logical error, so an error in the code.